test/quic_multistream_test.c

```diff
@@ -2081,15 +2089,18 @@ static int script_20_inject_plain(struct helper *h, QUIC_PKT_HDR *hdr,
         return 0;
 
     if (!TEST_true(WPACKET_quic_write_vlint(&wpkt, h->inject_word1)))
-        return 0;
+        goto err;
 
     if (!TEST_true(WPACKET_get_total_written(&wpkt, &written)))
-        return 0;
+        goto err;
 
     if (!qtest_fault_prepend_frame(h->qtf, frame_buf, written))
-        return 0;
+        goto err;
 
-    return 1;
+    ok = 1;
+err:
+    WPACKET_finish(&wpkt);
```

Comment:
Really we should be using WPACKET_cleanup() in the event of error and WPACKET_finish() for successful completion.

Here and below.